homeassistant/components/smartthings/smartapp.py

```diff
@@ -111,7 +112,11 @@ def get_webhook_url(hass: HomeAssistantType) -> str:
 
 
 def _get_app_template(hass: HomeAssistantType):
-    endpoint = f"at {hass.config.api.base_url}"
+    try:
+        endpoint = f"at {async_get_url(hass, allow_cloud=False, prefer_external=True)}"
```

Comment:
I think that this should be 
```suggestion
        endpoint = f"at {async_get_url(hass, allow_cloud=False, allow_internal=False)}"
```

This is used for SmartThings sending us webhooks.

Reply:
Actually it isn't. It is just a string for identification purposes.
(I figured identifying by external URL would be more recognizable).

The actual hooks use the Home Assistant webhook component, which has been patched as well in this PR.

The actual webhook creation: <https://github.com/home-assistant/core/blob/dfe2a457c05570518dc61847dc701e2f1349071f/homeassistant/components/smartthings/smartapp.py#L101>

We do need to add a helper and/or add more support to the webhooks in order to manage cloud vs non-cloud, and also handle the SSL requirement. That would be a nice followup for this PR I would say.